src/useKey/useKey.ts

```diff
@@ -0,0 +1,41 @@
+import { DependencyList, RefObject, useMemo } from 'react';
+import { useEventListener } from '..';
+import { noop, isBrowser } from '../util/const';
+
+export type IKeyPredicate = (event: KeyboardEvent) => boolean;
+export type IKeyFilter = null | undefined | string | IKeyPredicate;
+export type IHandler = (event: KeyboardEvent) => void;
+
+export type IUseKeyOptions<T extends EventTarget> = {
+  event?: 'keydown' | 'keypress' | 'keyup';
+  target?: RefObject<T> | T | null;
+  options?: boolean | AddEventListenerOptions;
```

Comment:
IMO it is better to name it `eventOptions`, since `options` filed within `options` object sounds a bit off for me🤪